### ISSUE TITLE
MiKTeX binaries do nothing under Cygwin and Windows 10 Ent 2019 LTSC

### ISSUE THREAD

**Issue by HGSqm** [MiKTeX/miktex#881]
- [x] I have read and acknowledged this HOWTO article:  
  [Submitting a MiKTeX problem report](https://miktex.org/howto/bug-report)

Running a binary like `pdflatex` or `pdfcrop` in a Cygwin bash does not work. After a second it returns to the prompt, nothing else happens. There is no entry in the log files.

Running `pdflatex` from a Windows Command Prompt works without any problems, as well as invoking it from TeXnicCenter.

I assume it has something to to with Windows. I was able to reproduce the issue on another machine running Windows 10 Enterprise LTSC 2019 as well, but not on a machine with Windows 10 Professional 20H2. MiKTeX and Cygwin are both up to date.

The first time I realized that problem was this spring. I used to use a bash script doing some modifiations to filenames before invoking `pdfcrop`. That stopped working, but I didn't investigate further at that time. I remember that I updated MiKTeX in April. The update before that was done maybe at the beginning of 2020. Because of a bigger project I didn't do any updates in between. Unfortunately I can't give more precise dates because I completely removed MiKTeX and did a fresh installation last week. And the old log files are lost.

I use the 32 bit version of MiKTeX and the 64 bit version of Cygwin. But as said above that combination works fine on another PC.

I tried to log what happens after calling pdflatex with Process Monitor. Merely at the end there was a call to WerFault.exe. Now I looked in event viewer of Windows and found an error concerning `ucrtbase.dll`.

```
Name der fehlerhaften Anwendung: pdflatex.exe, Version: 4.7.0.0, Zeitstempel: 0x60f2a707
Name des fehlerhaften Moduls: ucrtbase.dll, Version: 10.0.17763.1490, Zeitstempel: 0x51d4b57a
Ausnahmecode: 0xc0000409
Fehleroffset: 0x000a5b3b
ID des fehlerhaften Prozesses: 0x1b90
Startzeit der fehlerhaften Anwendung: 0x01d7820536487881
Pfad der fehlerhaften Anwendung: C:\Program Files (x86)\MiKTeX\miktex\bin\pdflatex.exe
Pfad des fehlerhaften Moduls: C:\Windows\System32\ucrtbase.dll
Berichtskennung: 41d35de7-fff5-4978-9a85-1329f5a48e3b
Vollständiger Name des fehlerhaften Pakets: 
Anwendungs-ID, die relativ zum fehlerhaften Paket ist: 
```

This DLL seems to be connected to the Visual C++ Redistributable. There are several different versions installed on this PC. But I don't know which one is necessary for MiKTeX.

[miktex-report.txt](https://github.com/MiKTeX/miktex/files/6877154/miktex-report.txt)

**Comment by edocevoli** [comment]
Thank you. I cannot reproduce this with 64-bit MiKTeX/Cygwin. It seems, that 32-bit MiKTeX executables cannot be started from 64-bit Cygwin. This should be documented as a known limitation. But it wont get fixed.

**Comment by HGSqm** [comment]
Thank you fo the quick investigation and reply. I already thought that you cannot reproduce this, but I don't think it's a 32-bit vs 64-bit issue. In that configuration it worked last year without problems. In the meantime I uninstalled the 32-bit MiKTeX and installed the 64-bit version. But that doesn't change the behaviour when running something from Cygwin.

And as I mentioned I have (at home) a PC running 32-bit MiKTeX and 64-bit Cygwin without problems. So documenting it like you said wouldn't be correct. By the way I also have the 32-bit version of Cygwin installed. But that doesn't help either.

I still think that it's an issue depending on Windows versions, as that is currently the only difference I see. But I understand that you won't fix it because the LTSC version of Windows is rarely used. It is not available for everyone.

I found that building MiKTeX requires Visual Studio 2015. So I checked the C++ Redistributable packages. 2015 is installed both 32-bit and 64-bit version.
